Spanner/src/Operation.php

```diff
@@ -436,12 +444,24 @@ public function transaction(Session $session, array $options = [])
         }
 
         if (!$options['singleUse']) {
-            $res = $this->beginTransaction($session, $options);
+            if (isset($options['begin']) and !isset($transactionOptions['partitionedDml'])) {
+                $res = ['id' => ['begin' => $options['begin']]];
+            } else {
+                $res = $this->beginTransaction($session, $options);
+            }
         } else {
             $res = [];
         }
 
-        return $this->createTransaction($session, $res, ['tag' => $transactionTag]);
+        if (isset($options['fromTransaction'])) {
+            return $res;
+        }
+
+        return $this->createTransaction(
+            $session,
+            $res,
+            ['tag' => $transactionTag, 'isRetry' => $options['isRetry'], 'options' => $options]
```

Comment:
Wasn't this a bug not sending the 'isRetry' status?
Ref: https://cloud.google.com/php/docs/reference/cloud-spanner/latest/Transaction#_Google_Cloud_Spanner_Transaction__isRetry__